hawkbit-security-integration/src/main/java/org/eclipse/hawkbit/security/PreAuthTokenSourceTrustAuthenticationProvider.java

```diff
@@ -118,6 +110,36 @@ public Authentication authenticate(final Authentication authentication) {
 
         throw new BadCredentialsException("The provided principal and credentials are not match");
     }
+    
+	/**
+	 * 
+	 * The credentials may either be of type HeaderAuthentication or of type
+	 * Collection<HeaderAuthentication> depending on the authentication mode in
+	 * use (the latter is used in case of trusted reverse-proxy). It is checked
+	 * whether principal equals credentials (respectively if credentials
+	 * contains principal in case of collection) because we want to check if
+	 * e.g. controllerId containing in the URL equals the controllerId in the
+	 * special header set by the reverse-proxy which extracted the CN from the
+	 * certificate.
+	 * 
+	 * @param principal
+	 * @param credentials
+	 * @param tokenDetails
+	 * @return
+	 */
+    private boolean calculateAuthenticationSuccess(Object principal, Object credentials, Object tokenDetails) {
+    	boolean successAuthentication = false;
+    	if (principal.equals(credentials)) {
```

Comment:
My intention was to perform the cheaper check in the first place (`principal.equals(credentials)`).
Otherwise there would always the `isAssignableFrom` check that I suspect to be more expensive.